modern-publish-plugin.md

```diff
@@ -0,0 +1,211 @@
+            id = "username"
+            name = "User Name"
+            url = "https://github.com/username/"
+        }
+    }
+  }
+
+  # Configure closeAndReleaseRepository
+  nexus {
+     baseUrl = "https://your_nexus_instance"
+     stagingProfile = "net.example"
+     # Start supporting the Gradle credentials API
+     respositoryUserName = "username"
+     respositoryPassword = "password"
+  }
+}
+```
+
+#### `com.vanniktech.maven.publish.java`
+
+Meant to be used with `java`, `java-library`, `org.jetbrains.kotlin.jvm`. 
+
+Pure Gradle configuration:
+```
+publishing {
+    publications {
+        create<MavenPublication>("maven") {
+            from(components["java"])
+        }
+    }
+}
+
+java {
+  withSourcesJar()
+  withJavadocJar()
+}
+```
+
+The plugin doesn't need to offer anything for sources and javadocs jars. It will
+simply create a publication using `components["java"]`.
+
+#### `com.vanniktech.maven.publish.gradle`
+
+Meant to be used with `java-gradle-plugin`.
+
+Pure Gradle configuration:
+```
+java {
+  withSourcesJar()
+  withJavadocJar()
+}
+```
+
+Gradle automatically adds a publication when both `java-gradle-plugin` and 
+`maven-publish` are applied. There is no need for our own APIs, so this plugin
+does not have any functionality of it's own.
+
+#### `com.vanniktech.maven.publish.android`
+
+Meant to be used with `com.android.library`.
+
+Pure Gradle configuration:
+```
+afterEvaluate {
+    publishing {
+        publications {
+            create<MavenPublication>("maven") {
+                from(components["release"])
+            }
+        }
+    }
+}
+```
+
+
+There is no API available to create source and javadoc jars for Android 
+projects. The plugin will provide an API to configure those as well as 
+which variant to publish. These properties/methods can be removed/deprecated once
+AGP offers it's own methods and adds multi variant publishing.
+
+```
+androidPublishCompat {
+  androidVariantToPublish = "release"
+  withSourcesJar()
+  withJavadocJar()
+}
+```
+
+#### `com.vanniktech.maven.publish.kotlin.js`
+
+Meant to be used with `org.jetbrais.kotlin.js`.
+
+Pure Gradle configuration:
+```
+TODO
+```
+
+TODO
+
+#### `com.vanniktech.maven.publish.kotlin.mpp`
+
+Meant to be used with `org.jetbrais.kotlin.multiplatform`.
+
+Pure Gradle configuration: n/a
+
+Like `java-gradle-plugin` the Kotlin mpp plugin sets up a publication 
+automatically. It also adds source jars automatically. This plugin adds
+an API to add javadoc publishing. There still is a `withSourcesJar()` which will
+add empty source jars to all publications missing one, this can be removed once
+the behavior is fixed on the Kotlin side.
+
+```
+kotlinMultiplatformPublishCompat {
+  withSourcesJar()
+  withJavadocJar()
+}
+```
+
+#### `com.vanniktech.maven.publish`
+
+- calls `publishToMavenCentral()`
+- sets `groupId`, `version` and `baseArtifactId` based on Gradle properties
+- configures `pom` based on Gradle properties
```

Comment:
Should we add a deprecation message when reading `gradle.properties` or do we want to support them long term here?

Reply:
We could output one warning for all of them. That way we encourage upgrading without spamming the entire console for every single property.